src/snowflake/snowpark/modin/plugin/_internal/binary_op_utils.py

```diff
@@ -300,6 +301,29 @@ def compute_binary_op_between_snowpark_columns(
     ):
         # string/string case (only for add)
         binary_op_result_column = concat(first_operand, second_operand)
+    elif (
+        op == "add"
+        and isinstance(first_datatype(), TimestampType)
+        and isinstance(second_datatype(), TimedeltaType)
```

Comment:
let's always check `TimedeltaType` first. I'm going to make a small refactoring on `create_lazy_type_functions` and `quoted_identifier_to_snowflake_type` to make sure if it just wants to get cached type, it does not need to call schema and make metadata query.

Reply:
> let's always check TimedeltaType first

Done.

> I'm going to make a small refactoring on create_lazy_type_functions and quoted_identifier_to_snowflake_type to make sure if it just wants to get cached type, it does not need to call schema and make metadata query.

Yep, I was thinking about the performance costs of these type checks and I was thinking of rewriting the type getters as well.

Reply:
I have a drat now https://github.com/snowflakedb/snowpark-python/pull/2159 FYI.